pkg/replication/replication.go

```diff
@@ -370,14 +409,18 @@ func (r *replication) ensureInReality(node types.NodeName, quitCh <-chan struct{
 
 func (r *replication) ensureHealthy(
 	node types.NodeName,
-	quitCh <-chan struct{},
+	timeoutCh <-chan struct{},
 	nodeLogger logging.Logger,
 	aggregateHealth *podHealth,
-) {
+) error {
 	for {
 		select {
-		case <-quitCh:
-			return
+		case <-r.quitCh:
+			return errQuit
+		case <-timeoutCh:
+			return errTimeout
+		case <-r.replicationCancelledCh:
```

Comment:
I think `replicationCancelled` is going to race with the quit scenario because `handleRenewalErrors` closes quit after a cancellation